stl/inc/sstream

```diff
@@ -100,6 +163,55 @@ public:
         _Init(_Newstr.c_str(), _Newstr.size(), _Mystate);
     }
 
+#if _HAS_CXX20
+    template <class _Alloc2, enable_if_t<_Is_allocator<_Alloc2>::value, int> = 0>
+    _NODISCARD basic_string<_Elem, _Traits, _Alloc2> str(const _Alloc2& _Al) const {
+        return basic_string<_Elem, _Traits, _Alloc2>{view(), _Al};
+    }
+
+    // The buffer cannot be moved to a string directly, because
+    // the buffer may already be full, and the terminating char is not '\0'.
+    // In that case, copy the string as usual.
+    _NODISCARD _Mystr str() && {
+        _Mystr _Result{_String_constructor_rvalue_allocator_tag{}, _STD move(_Al)};
+        const auto _View = _Get_buffer_view();
+        if (_View._Size >= _View._Res) {

```

Comment:
It seems like this condition should be impossible? If we're storing more characters than we have space for something is wrong :)

Reply:
_Size maybe equal to _Res. stringbuf doesn't allocate for the terminating zero.

Reply:
I see; then change >= to == and add a comment? Or maybe we should be adding the null here rather than in `_Move_assign_from_buffer` to make it obvious?